neqo-transport/src/connection/mod.rs

```diff
@@ -2155,20 +2162,30 @@ impl Connection {
             return (tokens, false, false);
         }
 
-        if primary {
-            if space == PacketNumberSpace::ApplicationData {
-                self.write_appdata_frames(builder, &mut tokens);
-            } else {
-                let stats = &mut self.stats.borrow_mut().frame_tx;
-                self.crypto.write_frame(space, builder, &mut tokens, stats);
+        if path.borrow().pmtud().borrow().is_probe_prepared() {
+            // If this is a PMTUD probe, don't include any other frames and record it as sent.
```

Comment:
Does this mean that we are displacing "real" frames with a probe?

Reply:
Yes. I'll change it to using payload data.